### ISSUE TITLE
[Enhancement] [2.0] Polished code blocks with copy button

### ISSUE THREAD

**Issue by Redsandro** [Zettlr/Zettlr#2369]
### Description

Make the code blocks look slightly more polished and add a copy button that appears on hover.

(Forked from #2243)

### Proposed Changes

1. If we find a way to mark the first and last line of a code block with a class so that we can apply a `border-radius-top` and `border-radius-bottom`, the `border-radius` of a code-block could be consistent with an inline code-tag.

Currently code-block looks like this:
![image](https://user-images.githubusercontent.com/1702193/131269458-03e9cfdb-c418-4d13-92e5-ae133233fb89.png)

Mockup of 4px border-radius:

![image](https://user-images.githubusercontent.com/1702193/131269757-3968e970-0fd6-461a-aa18-7105c1f9b79e.png)

2. If there can be a "copy" button on hover to copy entire code blocks to the clipboard, it will be helpful for system admins and other people that keep notes with terminal commands.

3. If we could css-select the code block language string separately from the backticks, we can style the language to be more clear, either in updated themes, and/or in custom css.

Mockup:
![image](https://user-images.githubusercontent.com/1702193/131270410-cdc38744-cbe9-494e-b60d-b04bf83e066a.png)

4. An option to hide the backticks while the cursor is not inside the code block is desired by some. This is discussed in #2243.

### Summary

- [ ] 1: Border-radius consistent with inline code-tag;
- [ ] 2: copy code button;
- [ ] 3: css-selector for code language string;
- [ ] 4: (option to) hide backticks when block is not being edited.

**Comment by nathanlesage** [comment]
> If we find a way to mark the first and last line of a code block with a class

That should be easy

> If there can be a "copy" button on hover to copy entire code blocks to the clipboard

Not as easy, but still possible

> If we could css-select the code block language string separately from the backticks

Extremely difficult. Right now we can make use of a plugin, but for that we would need to basically … or wait. Puh, it's quite the endeavour but we *could* copy the required code from the multiplexer into the main highlight mode due to the improvements of the past months. But still, would take a lot of time for a very small gain.

> An option to hide the backticks while the cursor is not inside the code block is desired by some.

Could be a little bit easier.

***

Summing up: These are actually four separate issues in one, three of which are definitely possible.

**Comment by Redsandro** [comment]
> These are actually four separate issues in one

🪄
